fake/fake.go

```diff
@@ -0,0 +1,117 @@
+/*
+Package fake provides utilities for generating fake data.
+*/
+package fake
+
+import (
+	"crypto/rand"
+	"fmt"
+	rnd "github.com/kashifkhan0771/utils/rand"
+	"io"
+	"time"
+)
+
+const (
+	EpochYear   = 1970
+	EpochMonth  = 1
+	EpochDay    = 1
+	EpochHour   = 0
+	EpochMinute = 0
+	EpochSecond = 0
+	EpochNano   = 0
+)
+
+const MaxStreetNumber = 150
+
+var streetNames = []string{"Main St", "High St", "Broadway", "Maple Ave", "Oak St", "Pine St", "Cedar St", "Elm St"}
+var cities = []string{"Springfield", "Rivertown", "Lakeview", "Greenville", "Fairview", "Madison", "Georgetown", "Clinton"}
+var states = []string{"AL", "AK", "AZ", "AR", "CA", "CO", "CT", "DE", "FL", "GA", "HI", "ID", "IL", "IN", "IA", "KS", "KY", "LA", "ME", "MD", "MA", "MI", "MN", "MS", "MO", "MT", "NE", "NV", "NH", "NJ", "NM", "NY", "NC", "ND", "OH", "OK", "OR", "PA", "RI", "SC", "SD", "TN", "TX", "UT", "VT", "VA", "WA", "WV", "WI", "WY"}
+var postalCodes = []string{"12345", "23456", "34567", "45678", "56789", "67890", "78901", "89012", "90123"}
+
+// RandomUUID generates a fake UUIDv4.
+func RandomUUID() (string, error) {
+	uuid := make([]byte, 16)
+	_, err := io.ReadFull(rand.Reader, uuid)
+	if err != nil {
+		return "", err
+	}
+
+	// Set the version to 4
+	uuid[6] = (uuid[6] & 0x0f) | 0x40
+	// Set the variant to 2 (RFC 4122)
+	uuid[8] = (uuid[8] & 0x3f) | 0x80
+
+	return fmt.Sprintf("%08x-%04x-%04x-%04x-%12x", uuid[:4], uuid[4:6], uuid[6:8], uuid[8:10], uuid[10:]), nil
+}
+
+// RandomDate generates a random date.
+func RandomDate() (time.Time, error) {
+	start := time.Date(EpochYear, time.Month(EpochMonth), EpochDay, EpochHour, EpochMinute, EpochSecond, EpochNano, time.UTC).Unix()
+	end := time.Now().Unix()
+
+	sec, err := rnd.NumberInRange(start, end)
+	if err != nil {
+		return time.Time{}, err
+	}
+
+	nsec, err := rnd.NumberInRange(0, 1e9)
+	if err != nil {
+		return time.Time{}, err
+	}
+
+	return time.Unix(sec, nsec), nil
+}
+
+// RandomPhoneNumber generates a random phone number.
+func RandomPhoneNumber() (string, error) {
+	areaCode, err := rnd.NumberInRange(100, 999)
```

Comment:
Area codes can start from `01` I guess, e.g Pakistan as +92 area code.

Reply:
The function generates only US phone numbers with the format: +1 (XXX) XXX-XXXX, and the area code in this context is (XXX). Generating phone numbers from all around the world would require a much more complex logic because, for example, the valid German phone number is: +49 30 901820

Reply:
Alright, I thought we are doing for every country. That's OK for now, maybe we can come later and implement more.